### PR TITLE
Adding posts model

### PR DESCRIPTION
**Pull request by johnathananderson**
Created a posts model for the Uplift backend to use for fitness expert posts

### CODE REVIEW

File: src/app/models/post.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+from . import *
+
+class Post(Base):
```
Comment (meganle) [inline src/app/models/post.py]: we should break these models up into their own files

File: src/app/models/post.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  __tablename__ = "post"
+
+  id = db.Column(db.Integer, primary_key=True)
+  large_picture = db.Column(db.String(200), nullable=False)
```
Comment (meganle) [inline src/app/models/post.py]: let's alphabetize all of the fields for each model, but id can stay as the first one

File: src/app/models/_all.py [hunk 3]
```diff
@@ -2,11 +2,27 @@
...
 
+class PostSchema(ModelSchema):
+  class Meta(ModelSchema.Meta):
+    model = Post
+
+class SocialMediaSchema(ModelSchema):
+  class Meta(ModelSchema.Meta):
+    model = SocialMedia
+
+class RoutineSchema(ModelSchema):
```
Comment (young-k) [inline src/app/models/_all.py]: this file should be alphabetized (`UserSchema` at the bottom)

File: src/app/models/_all.py [hunk 4]
```diff
@@ -2,11 +2,27 @@
...
+  class Meta(ModelSchema.Meta):
+    model = Routine
+
 # Serializers
-user_schema = UserSchema()
\ No newline at end of file
+user_schema = UserSchema()
+post_schema = PostSchema()
+social_media = SocialMediaSchema()
+routine = RoutineSchema()
```
Comment (young-k) [inline src/app/models/_all.py]: same comment here

File: src/app/models/_all.py [hunk 5]
```diff
@@ -2,11 +2,27 @@
 from marshmallow import fields
 
 from src.app.models.user import *
+from src.app.models.post import *
```
[the remark this answers is not in the document]
Comment (williambaibai) [reply src/app/models/_all.py]: +1

File: src/app/models/_all.py (hunk 3 above)
Comment (williambaibai) [reply src/app/models/_all.py]: Alphabetize schemas

File: src/app/models/_all.py (hunk 4 above)
Comment (williambaibai) [reply src/app/models/_all.py]: ^^

File: src/app/models/post.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
+from . import *
+
+class Post(Base):
+  __tablename__ = "post"
```
Comment (williambaibai) [inline src/app/models/post.py]: for consistency, let's keep the `tablename` plural, so `posts`

File: src/app/models/post.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  linkedin = db.Column(db.String(200), nullable=True)
+  website = db.Column(db.String(200), nullable=True)
+  post_id = db.Column(db.Integer, db.ForeignKey('post.id'), nullable=False)
+
+  def __init__(self, **kwargs):
+    # TODO: Implement this
+    pass
+
+class Routine(Base):
+  __tablename__ = "routine"
```
Comment (williambaibai) [inline src/app/models/post.py]: `routines`

File: src/app/models/post.py [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+  name = db.Column(db.String(100), nullable=False)
+  biography = db.Column(db.String(300), nullable=False)
+  social_media = db.relationship('SocialMedia', backref='post')
+  expertises = db.Column(db.String(200), nullable=False)
+  suggested_routines = db.relationship('Routine', backref='post')
+  summary = db.Column(db.String(300), nullable=False)
+
+  def __init__(self, **kwargs):
+    # TODO: Implement this
+    pass
```
Comment (cswenberg) [inline src/app/models/post.py]: should we be merging in incomplete code to master? @meganle @young-k.  I think it'd be best to finish up the __init__ methods

File: src/app/models/post.py [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    self.biography = kwargs.get('biography', '')
+    self.expertises = kwargs.get('expertises', '')
+    self.large_picture = kwargs.get('large_picture', '')
+    self.name = kwargs.get('name', '')
+    self.small_picture = kwargs.get('small_picture', '')
+    self.summary = kwargs.get('summary', '')
+
+  def serialize(self):
+        return {
+            'id': self.id,
```
Comment (williambaibai) [inline src/app/models/post.py]: Indentations are off for all `serialize method

File: src/app/models/routine.py [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+  title = db.Column(db.String(100), nullable=False)
+
+  def __init__(self, **kwargs):
+    self.category = kwargs.get('category', '')
+    self.post_id = kwargs.get('post_id')
+    self.steps = kwargs.get('steps', '')
+    self.title = kwargs.get('title', '')
+
+  def serialize(self):
+        return {
```
Comment (williambaibai) [inline src/app/models/routine.py]: here too

File: src/app/models/social_media.py [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+  def __init__(self, **kwargs):
+    self.facebook = kwargs.get('facebook', '')
+    self.instagram = kwargs.get('instagram', '')
+    self.linkedin = kwargs.get('linkedin', '')
+    self.post_id = kwargs.get('post_id')
+    self.twitter = kwargs.get('twitter', '')
+    self.website = kwargs.get('website', '')
+
+  def serialize(self):
+        return {
```
Comment (williambaibai) [inline src/app/models/social_media.py]: and here

File: src/app/models/post.py [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+
+  id = db.Column(db.Integer, primary_key=True)
+  biography = db.Column(db.String(300), nullable=False)
+  expertises = db.Column(db.String(200), nullable=False)
+  large_picture = db.Column(db.String(200), nullable=False)
+  name = db.Column(db.String(100), nullable=False)
+  small_picture = db.Column(db.String(200), nullable=False)
+  social_media = db.relationship('SocialMedia', backref='posts')
+  suggested_routines = db.relationship('Routine', backref='posts')
+  summary = db.Column(db.String(300), nullable=False)
```
Comment (cswenberg) [inline src/app/models/post.py]: maybe consider making these string lengths their own constants like short, medium, long.

Comment (CameronHamidi) [reply src/app/models/post.py]: Looks good! But could you add another column for a personal quote?

**Pull request merged**